src/NodaTime.Demo/LocalDateDemo.cs

```diff
@@ -104,5 +106,82 @@ public void Min()
             LocalDate min = Snippet.For(LocalDate.Min(earlyJune, lateJune));
             Assert.AreEqual(new LocalDate(2010, 6, 5), min);
         }
+
+        [Test]
+        public void FromDateTime()
+        {
+            DateTime earlyJune = new DateTime(2010, 6, 5);
+            LocalDate date = Snippet.For(LocalDate.FromDateTime(earlyJune));
+            Assert.AreEqual(new LocalDate(2010, 6, 5), date);
+        }
+
+        [Test]
+        public void FromDateTimeWithCalendarSystem()
+        {
+            DateTime earlyJune = new DateTime(2010, 6, 5);
+            CalendarSystem calendar = CalendarSystem.ForId("Julian");
+            LocalDate date = Snippet.For(LocalDate.FromDateTime(earlyJune, calendar));
+            //Until 2099, Julian calendar is 13 days behind Gregorian
+            Assert.AreEqual(2010, date.Year);
+            Assert.AreEqual(5, date.Month);
+            Assert.AreEqual(23, date.Day);
+        }
+
+        [Test]
+        public void CommonEra()
+        {
+            LocalDate date = Snippet.For(new LocalDate(Era.Common, 2010, 6, 16));
+            Assert.AreEqual(new LocalDate(2010, 6, 16), date);
+        }
+
+        [Test]
+        public void BeforeCommonEra()
+        {
+            LocalDate date = Snippet.For(new LocalDate(Era.BeforeCommon, 2010, 6, 16));
+            Assert.AreEqual(new LocalDate(Era.BeforeCommon, 2010, 6, 16), date);
+        }
+
+        [Test]
+        public void FromWeekYearWeekAndDay()
+        {
+            LocalDate date = Snippet.For(LocalDate.FromWeekYearWeekAndDay(2010,24,IsoDayOfWeek.Wednesday));
+            Assert.AreEqual(new LocalDate(2010, 6,16), date);
+        }
+
+        [Test]
+        public void FromYearMonthWeekAndDay()
+        {
+            LocalDate date = Snippet.For(LocalDate.FromYearMonthWeekAndDay(2010, 6, 3, IsoDayOfWeek.Wednesday));
+            Assert.AreEqual(new LocalDate(2010, 6, 16), date);
+        }
+
+        [Test]
+        public void ToYearMonth()
+        {
+            YearMonth yearMonth = Snippet.For(new LocalDate(2010, 6, 16).ToYearMonth());
+            Assert.AreEqual(new YearMonth(2010,6), yearMonth);
+        }
+
+        [Test]
+        public void Plus()
+        {
+            LocalDate date = Snippet.For(new LocalDate(2010, 6, 16).Plus(Period.FromDays(1)));
+            Assert.AreEqual(new LocalDate(2010, 6, 17), date);
+        }
+
+        [Test]
+        public void Subtract()
+        {
+            LocalDate date = Snippet.For(LocalDate.Subtract(new LocalDate(2010, 6, 16), Period.FromDays(1)));
```

Comment:
I would suggest subtracting from "the result of the Plus example" - in particular, if you add a month to January 30th 2010, you get February 28th 2010... and subtracting a month from that *doesn't* get you back to January 30th...